src/Packetery/Module/WpdbAdapter.php

```diff
@@ -0,0 +1,299 @@
+	 *
+	 * @return int|false The number of rows inserted, or false on error.
+	 */
+	public function insert( string $table, array $data ) {
+		$result = $this->wpdb->insert( $table, $data );
+		if ( false === $result && '' !== $this->getLastWpdbError() && $this->isPacketeryTableQueried( (string) $this->wpdb->last_query ) ) {
+			$this->handleError( $this->getLastWpdbError() );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Updates a row in the table.
+	 *
+	 * @param string $table Table name.
+	 * @param array  $data  Data to update (in column => value pairs).
+	 * @param array  $where A named array of WHERE clauses (in column => value pairs).
+	 *
+	 * @return int|false The number of rows updated, or false on error.
+	 */
+	public function update( string $table, array $data, array $where ) {
+		$result = $this->wpdb->update( $table, $data, $where );
+		if ( false === $result && '' !== $this->getLastWpdbError() && $this->isPacketeryTableQueried( (string) $this->wpdb->last_query ) ) {
+			$this->handleError( $this->getLastWpdbError() );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Gets charset collate.
+	 *
+	 * @return string
+	 */
+	public function get_charset_collate(): string {
+		return $this->wpdb->get_charset_collate();
+	}
+
+	/**
+	 * Retrieves an entire SQL result set from the database (i.e., many rows).
+	 *
+	 * @param string $query  SQL query.
+	 * @param string $output Optional. Any of ARRAY_A | ARRAY_N | OBJECT | OBJECT_K constants.
+	 *
+	 * @return array|object|null Database query results.
+	 */
+	public function get_results( string $query, string $output = OBJECT ) {
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+		$result = $this->wpdb->get_results( $query, $output );
+		if ( '' !== $this->getLastWpdbError() && $this->isPacketeryTableQueried( (string) $this->wpdb->last_query ) ) {
+			$this->handleError( $this->getLastWpdbError() );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Retrieves one variable from the database.
+	 *
+	 * @param string $query SQL query. Defaults to null, use the result from the previous query.
+	 *
+	 * @return string|null Database query result (as string), or null on failure.
+	 */
+	public function get_var( string $query ): ?string {
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+		$result = $this->wpdb->get_var( $query );
+		if ( null === $result && '' !== $this->getLastWpdbError() && $this->isPacketeryTableQueried( (string) $this->wpdb->last_query ) ) {
+			$this->handleError( $this->getLastWpdbError() );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Tells if packetery table is queried.
+	 *
+	 * @param string $query Query.
+	 *
+	 * @return bool
+	 */
+	private function isPacketeryTableQueried( string $query ): bool {
+		$matches = [];
+		$prefix  = $this->getPacketeryPrefix();
+		preg_match( '~\s*(FROM|JOIN|INTO|UPDATE|TABLE)\s*`?(' . preg_quote( $prefix, '~' ) . ')~i', $query, $matches );
+
+		return array_pop( $matches ) === $prefix;
```

Comment:
Nebylo by lepší porovnat návratovou hodnotu preg_match s 1?
preg_match() returns 1 if the pattern matches given subject, 0 if it does not, or false on failure.